sklearn/ensemble/_hist_gradient_boosting/tests/test_gradient_boosting.py

```diff
@@ -98,40 +100,58 @@ def test_early_stopping_regression(scoring, validation_fraction,
                         random_state=0)
 ))
 @pytest.mark.parametrize(
-    'scoring, validation_fraction, n_iter_no_change, tol', [
-        ('accuracy', .1, 5, 1e-7),  # use scorer
-        ('accuracy', None, 5, 1e-1),  # use scorer on training data
-        (None, .1, 5, 1e-7),  # same with default scorerscor
-        (None, None, 5, 1e-1),
-        ('loss', .1, 5, 1e-7),  # use loss
-        ('loss', None, 5, 1e-1),  # use loss on training data
-        (None, None, None, None),  # no early stopping
+    'scoring, validation_fraction, early_stopping, n_iter_no_change, tol', [
+        ('accuracy', .1, True, 5, 1e-7),  # use scorer
+        ('accuracy', None, True, 5, 1e-1),  # use scorer on training data
+        (None, .1, True, 5, 1e-7),  # same with default scorer
+        (None, None, True, 5, 1e-1),
+        ('loss', .1, True, 5, 1e-7),  # use loss
+        ('loss', None, True, 5, 1e-1),  # use loss on training data
+        (None, None, False, 5, None),  # no early stopping
         ])
 def test_early_stopping_classification(data, scoring, validation_fraction,
-                                       n_iter_no_change, tol):
+                                       early_stopping, n_iter_no_change, tol):
 
     max_iter = 50
 
     X, y = data
 
     gb = HistGradientBoostingClassifier(
-        verbose=1,  # just for coverage
+        verbose=0,  # just for coverage
         min_samples_leaf=5,  # easier to overfit fast
         scoring=scoring,
         tol=tol,
+        early_stopping=early_stopping,
         validation_fraction=validation_fraction,
         max_iter=max_iter,
         n_iter_no_change=n_iter_no_change,
         random_state=0
     )
     gb.fit(X, y)
 
-    if n_iter_no_change is not None:
+    if early_stopping is True:
         assert n_iter_no_change <= gb.n_iter_ < max_iter
     else:
         assert gb.n_iter_ == max_iter
 
 
+@pytest.mark.parametrize('GradientBoosting, X, y', [
+    (HistGradientBoostingClassifier, X_classification, y_classification),
+    (HistGradientBoostingClassifier, *make_classification(n_samples=10001)),
+    (HistGradientBoostingRegressor, X_regression, y_regression),
+    (HistGradientBoostingRegressor, *make_regression(n_samples=10001))
+])
+def test_early_stopping_default(GradientBoosting, X, y):
```

Comment:
Yes, it is a much better idea than what I did! Something like [sklearn.datasets.make_hastie_10_2](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_hastie_10_2.html), but more adapted to a tree-based algorithm?

Reply:
If hastie works then perfect, but feel free to also manually create the datasets, e.g.

```py
X = rng.normal(size=(n_samples, 1))
y = X > 0
```

That could probably even work for both classification and regression

Reply:
Do you mind if I increase the tolerance? Even with your dumb dataset and `max_iter=100`the test fails...

Reply:
Sure, whatever it takes to make the ES happen as long as it makes sense. We don't really care about any accuracy here, we just want to make sure that the ES happens

Reply:
> Do you mind if I increase the tolerance? Even with your dumb dataset and max_iter=100the test fails...

If the dataset is very easy, you can increase the learning rate to 0.5 or even 1 to make it converge in fewer iterations. Otherwise it might still underfit after 5 trees.